dask_kubernetes/operator/core.py

```diff
@@ -0,0 +1,88 @@
+# import asyncio
+# import subprocess
+
+from distributed.deploy import Cluster
+from distributed.core import rpc
+
+# import kopf
+import kubernetes
+
+from dask_kubernetes.utils import (
+    #     namespace_default,
+    get_external_address_for_scheduler_service,
+    #     check_dependency,
+)
+
+from daskcluster import (
+    build_cluster_spec,
+)  # , wait_for_scheduler, build_worker_group_spec
+
+
+class KubeCluster2(Cluster):
+    """Launch a Dask Cluster on Kubernetes using the Operator"""
+
+    def __init__(
+        self,
+        name,
+        namespace="default",
+        image="daskdev/dask:latest",
+        replicas=3,
+        resources={},
+        env={},
+        loop=None,
+        asynchronous=False,
+        **kwargs
+    ):
+        self.name = name
+        self.namespace = namespace
+        self.core_api = None
+        self.custom_api = None
+        self.image = image
+        self.replicas = replicas
+        self.resources = resources
+        self.env = env
+
+        super().__init__(asynchronous=asynchronous, **kwargs)
+        if not self.asynchronous:
+            self._loop_runner.start()
+            self.sync(self._start)
```

Comment:
This is failing with `NewConnectionError`. Has this happened to you before @jacobtomlinson?

Reply:
I don't think so. Which module does`NewConnectionError` come from?

Reply:
Hmm no idea. Perhaps this is happening somewhere inside `_start` and by running it async like this is causing the traceback to report the wrong line. This could well be related to the `kubernetes` API calls.